conn.go

```diff
@@ -91,7 +99,7 @@ func (client *Client) DialContext(ctx context.Context, network, address string)
 		return nil, fmt.Errorf("could not authenticate with ZK server: %w", err)
 	}
 
-	go c.handleReads()
+	go c.handleReadWrites()
```

Comment:
Why not handle the writes in the same Goroutine where we were sending the keepalives?

Reply:
because it must come before the read, calling it in the `read` thread makes sure of that.

Reply:
And why is that? It is not how things are done in this lib: https://github.com/go-zookeeper/zk

Reply:
I changed handerReads to also do the writes (before the reads), the API calls send write requests via channel to handleReadWrites go routine instead of writing directly to the network connection. This way there is one thread that does the io calls sequentially.